modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorSwatch.ts

```diff
@@ -187,19 +187,22 @@ const registerTextColorMenuItem = (editor: Editor, name: string, format: string,
 };
 
 const colorPickerDialog = (editor: Editor) => (callback, value: string) => {
-  const getOnSubmit = (callback) => (api) => {
+  let isValid = false;
+
+  const getOnSubmit = (callback) => (api: Dialog.DialogInstanceApi<ColorSwatchDialogData>) => {
```

Comment:
Since you've added some types, could you also add one for callback here? I think it should be this:
```suggestion
  const getOnSubmit = (callback: ColorInputCallback) => (api: Dialog.DialogInstanceApi<ColorSwatchDialogData>) => {
```
To do that though, you probably need to move the `ColorInputCallback` type from `ColorInputBackstage` to here and then export it. So up to you, as it is adding scope :shrug: